lib/ruby_lsp/store.rb

```diff
@@ -22,25 +22,23 @@ def initialize
 
     sig { params(uri: URI::Generic).returns(Document) }
     def get(uri)
-      path = uri.to_standardized_path
-      return T.must(@state[T.must(uri.opaque)]) unless path
-
-      document = @state[path]
+      document = @state[uri.to_s]
       return document unless document.nil?
 
-      set(uri: uri, source: File.binread(CGI.unescape(path)), version: 0)
-      T.must(@state[path])
+      path = T.must(uri.to_standardized_path)
```

Comment:
Would it make sense to change this to `to_standardized_uri` and handle URIs everywhere?

Reply:
Depending on the request we use either an URI or a path, this could be confusing.

I was wondering if we shouldn't always use URIs everywhere and provides services to read files etc. from URIs.

Reply:
Indeed, depending on what's happening the LSP needs a file path or a URI. I'm not sure if we could further standardize this, but it would be a significant refactor.

I created #896 to explore that, but I'd like to move forward with the bug fix first since it's breaking fundamental parts of the LSP.